chains/ethereum/runner/runner.go

```diff
@@ -104,11 +113,31 @@ func NewRunner(ctx context.Context, logger *zap.Logger, spec loadtesttypes.LoadT
 		sentTxs:         make([]*inttypes.SentTx, 0, 100),
 		blocksProcessed: 0,
 		nonces:          &nonces,
+		senderToWallet:  senderToWallet,
 	}
 
 	return r, nil
 }
 
+// getWalletForTx returns the appropriate wallet for sending a transaction based on the sender address.
+// This ensures each sender consistently uses the same endpoint.
+func (r *Runner) getWalletForTx(tx *gethtypes.Transaction) (*wallet.InteractingWallet, error) {
```

Comment:
kinda feel like making this return an error resulted in some ugly conditionals where this is used. in reality, this error should never occur, and if it does i dont think the program is even functional anymore? maybe we should just panic instead